servers/wanaku-server-quarkus/src/main/java/ai/wanaku/server/quarkus/api/v1/resources/ResourcesResource.java

```diff
@@ -58,6 +61,8 @@ public Response expose(ResourceReference resource) {
     @Path("/list")
     @GET
     @Produces(MediaType.APPLICATION_JSON)
+    @APIResponse(responseCode = "200", content = @Content(
+            schema = @Schema(type = SchemaType.ARRAY, implementation = ResourceReference.class)))
```

Comment:
I think using `RestResponse`, automatically generates the return type.

```java
import org.jboss.resteasy.reactive.RestResponse;

public RestResponse<InvoiceEntity> get(@PathParam("companyId") String companyId, @PathParam("id") Longid) { }
```

Reply:
RestResponse can help, but the API should be changed, in particular, right now we are returning a list in case of 200ok, and a string in case of error, this is not possible with `RestResponse<List<ResourceReference>>`, I cannot return `RestResponse.ResponseBuilder.serverError().entity("Failed to list resources").build()` in case of exception

Reply:
My .2 cents: better change the API now to make it adequate than later when we have to worry with breaking things for a larger group of users. So I am OK with this, as far as I am concerned.

Reply:
+1, I faced the same problem with ToolsList, missing the info in the generated client API

Reply:
so I did the same for tools here: https://github.com/wanaku-ai/wanaku/pull/113/files#diff-c649126c7ff75bbfd15343a50f30fe0cf7b9fffac107a7eaa5874d3d35029e05